src/kontrol/prove.py

```diff
@@ -1409,3 +1424,58 @@ def _final_term(empty_config: KInner, program: KInner, config_type: ConfigType)
             KVariable('STORAGESLOTLIST_FINAL'),
         ],
     )
+
+
+def _process_external_library_references(contract: Contract, foundry_contracts: dict[str, Contract]) -> list[KInner]:
+    """Create a list of KInner accounts for external libraries used in the given contract.
+
+    This function identifies external library placeholders within the contract's deployed bytecode,
+    deploys the required external libraries at random addresses, replaces the placeholders with the
+    actual addresses of the deployed libraries, and returns a list of KEVM account cells representing
+    the deployed external libraries.
+
+    :param contract: The contract object containing the deployed bytecode and external library references.
+    :param foundry_contracts: A dictionary mapping library names to Contract instances, representing all
+                             available contracts including external libraries.
+    :raises ValueError: If an external library referenced in the contract's bytecode is not found in foundry_contracts.
+    :return:  A list of KEVM account cells representing the deployed external libraries.
+    """
+    external_libs: list[KInner] = []
+    address_list: dict[str, str] = {}
+
+    for lib, ref_locations in contract.external_lib_refs.items():
+        ref_contract = foundry_contracts.get(lib)
+        if ref_contract is None:
+            raise ValueError(f'External library not found: {lib}')
+
+        if lib not in address_list:
+            new_address_hex = hash_str(ref_contract.deployed_bytecode)[:40].ljust(40, '0')
```

Comment:
Out of curiosity: wouldn't 
```
hash_str(ref_contract.deployed_bytecode)[:40].zfill(40)
```
work here? I just saw it being used later in the code too.

Reply:
`zfill` automatically adds `0` to the left, whereas `just` will add the given char to the right-hand side. Both options work in this case.